paper/paper.md

```diff
@@ -0,0 +1,354 @@
+style.](binary_different_units.png)
+
+Finally, in carefully comparing the output of scripts using `Pint`,
+`astropy.units`, and `unyt`, we found that in-place operations making use of a
+NumPy ufunc will unexpectedly strip units. For example, if `a` and `b` are
+`Pint` `Quantity` instances, `np.add(a, b, out=out))` will operate on `a` and
+`b` as if neither have units attached. Interestingly, without the `out` keyword,
+`Pint` does get the correct answer, so it's possible that this is a bug in
+`Pint` which we have reported upstream (see
+https://github.com/hgrecco/pint/issues/644).
+
+## Performance Comparison
+
+Checking units will always add some overhead over using hard-coded unit
+conversion factors. Thus a library that is entrusted with checking units in an
+application should incur the minimum possible overhead to avoid triggering
+performance regressions after integrating unit checking into an
+application. Optimally, a unit library will add zero overhead regardless of the
+size of the array. In practice that is not the case for any of the three
+libraries under consideration, and there is a minimum array size above which the
+overhead of doing a mathematical operation exceeds the overhead of checking
+units. It is thus worth benchmarking unit libraries in a fair manner, comparing
+with the same operation implemented using plain NumPy.
+
+Here we present such a benchmark. We made use of the `perf` [@perf] Python
+benchmarking tool, which not only provides facilities for establishing the
+statistical significance of a benchmark run, but also can tune a linux system to
+turn off operating system and hardware features like CPU throttling that might
+introduce variance in a benchmark. We made use of a Dell Latitude E7270 laptop
+equipped with an Intel i5-6300U CPU clocked at 2.4 Ghz. The testing environment
+was based on `Python 3.6.3` and had `NumPy 1.14.2`, `sympy 1.1.1`, `fastcache
+1.0.2`, `Astropy 3.0.1`, and `Pint 0.8.1` installed. `fastcache` [@fastcache] is
+an optional dependency of `sympy` that provides an optimized LRU cache
+implemented in C that can substantially speed up `sympy`. The system was
+instrumented using `perf system tune` to turn off CPU features that might
+interfere with stable benchmarks. We did not make any boot-time Linux kernel
+parameter changes.
+
+![A benchmark comparing the time to perform various binary arithmetic
+operations on input operands that have the same units. See Figure 1 for a
+detailed explanation of the plot style.](binary_same_units.png)
+
+For each of the benchmarks below, we show the ratio of the time to perform an
+operation with one of `unyt`, `Pint`, and `astopy.units`, $T_{\rm package}$, to
+the time it takes for NumPy to perform the equivalent operation, $T_{\rm
+numpy}$. For example, for the comparison of the performance of `np.add(a, b)`
+where `a` and `b` have different units with the same dimension, the
+corresponding benchmark to generate `T_{\rm numpy}` would use the code
```

Comment:
I think you intended `T_{\rm numpy}` to be rendered as LaTeX here instead of monospace.